src/common/Utils.ts

```diff
@@ -42,4 +43,13 @@ export default class Utils {
     );
     return nonUndefinedOwnedProps;
   }
+
+  /**
+   * @return Current timestamp as BigNumber object.
+   */
+  public static getCurrentTimestamp(): BigNumber {
```

Comment:
The function name should be improved. It's unclear it returns in milliseconds or in seconds.
The documentation is missing an explanation (it is in seconds, time represented in UTC).

Should we instead use `Date.now()` ?

Reply:
I have updated the implementation to use last block timestamp.